### DISCUSSION TITLE
ODE control loop

### DISCUSSION THREAD

**Discussion by ShuffleWire** [cyberbotics/webots#4305] [Q&A]
I'm working with a motor in "speed control" mode, and I've some issue to get the correct output speed.

As a consequence, I'm change to a force control, and added a custom PID control on it to get the speed correct. It work well.

However, The doc https://cyberbotics.com/doc/reference/motor?version=master#force-and-torque-control say thing about "ODE internal control loop". I'm expecting it's some sort of PID ? If so how could I change the parameters of this control loop ?

**Comment by omichel** [comment]
I don't think it is a PID, this is something internal to ODE for velocity control only. You should refer to ODE documentation and/or source code for more details.

**Reply by ShuffleWire** [reply]
yes you right, some reading remind me that there are no PID as involved. But still, It often don"ent work for me... TBF, I've a complex arm with loop, so it could be difficult anyway...
